app/Controller/FeedsController.php

```diff
@@ -190,262 +197,268 @@ public function importFeeds()
-                $this->request->data['Feed']['input_source'] = 'network';
-            }
-            if (!isset($this->request->data['Feed']['delete_local_file'])) {
-                $this->request->data['Feed']['delete_local_file'] = 0;
-            }
-            $this->request->data['Feed']['settings'] = json_encode($this->request->data['Feed']['settings']);
-            $this->request->data['Feed']['event_id'] = !empty($this->request->data['Feed']['fixed_event']) ? $this->request->data['Feed']['target_event'] : 0;
-            if (!$error) {
-                $result = $this->Feed->save($this->request->data);
-                if ($result) {
-                    $message = __('Feed added.');
-                    if ($this->_isRest()) {
-                        $feed = $this->Feed->find('first', array('conditions' => array('Feed.id' => $this->Feed->id), 'recursive' => -1));
-                        return $this->RestResponse->viewData($feed, $this->response->type());
+                $feed['Feed']['default'] = '0';
+                if (!isset($feed['Feed']['source_format'])) {
+                    $feed['Feed']['source_format'] = 'freetext';
+                }
+                if (!empty($feed['Feed']['source_format']) && ($feed['Feed']['source_format'] == 'misp')) {
+                    if (!empty($feed['Feed']['orgc_id'])) {
+                        $feed['Feed']['orgc_id'] = '0';
+                    }
+                }
+                if ($feed['Feed']['source_format'] == 'freetext') {
+                    if ($feed['Feed']['fixed_event'] == 1) {
+                        if (!empty($feed['Feed']['target_event']) && is_numeric($feed['Feed']['target_event'])) {
+                            $feed['Feed']['event_id'] = $feed['Feed']['target_event'];
+                        }
                     }
-                    $this->Flash->success($message);
-                    $this->redirect(array('controller' => 'feeds', 'action' => 'index'));
+                }
+                if (!isset($feed['Feed']['settings'])) {
+                    $feed['Feed']['settings'] = array();
                 } else {
-                    $message = __('Feed could not be added. Reason: %s', json_encode($this->Feed->validationErrors));
-                    if ($this->_isRest()) {
-                        return $this->RestResponse->saveFailResponse('Feeds', 'add', false, $message, $this->response->type());
+                    if (!empty($feed['Feed']['settings']['common']['excluderegex']) && !$this->__checkRegex($feed['Feed']['settings']['common']['excluderegex'])) {
+                        $regexErrorMessage = __('Invalid exclude regex. Make sure it\'s a delimited PCRE regex pattern.');
+                        if (!$this->_isRest()) {
+                            $this->Flash->error($regexErrorMessage);
+                            return true;
+                        } else {
+                            return new CakeResponse(array(
```

Comment:
`RestResponseComponent->saveFailResponse` could be used instead, unless you do it that way for a specific reason.